doc/source/basics/runtime_extension.rst

```diff
@@ -0,0 +1,123 @@
+.. _basics-runtime-extension:
+
+:octicon:`plug` Runtime Extension
+=================================
+
+fairseq2 provides a flexible runtime extension system that allows you to extend its functionality without modifying the core codebase.
+It uses ``setuptools`` entry points to dynamically load and register extensions during initialization.
+
+Overview
+--------
+
+The extension system is built around a dependency injection container (learn more in :ref:`basics-design-philosophy`) that manages fairseq2's components.
+Through this system, you can:
+
+* Register new models and model cards
+* Add custom asset providers and validators
+* Extend the runtime context
+* And more...
+
+Basic Usage
+-----------
+
+Before using any fairseq2 APIs, you must initialize the framework with :meth:`fairseq2.init_fairseq2`:
```

Comment:
This is not really needed anymore. Internal APIs call `init_fairseq2` if it has not been called so far. Calling `init_fairseq2` explicitly is mainly useful if you have a standalone non-recipe Python script where you mainly use fairseq2 and want to extend its dependency graph with some extra objects. Then at the beginning of your script you can do something like:

```
def register_my_script_objects(container: DependencyContainer) -> None:
   container.register(...)

if __name__ == "__main__":
  init_fairseq2(extras=register_my_script_objects)

  ...
```

Otherwise, as a user you can mostly forget about `init_fairseq2`.